### PR TITLE
Upgrade 3rd party dependencies

### PR DESCRIPTION
**Pull request by vidurananayakkara**
## Purpose
Upgrade 3rd party dependencies

## Goals
Upgrade 3rd party dependencies

## Approach
Upgrade 3rd party dependencies

## User stories
N/A

## Release note
N/A

## Documentation
N/A - This PR does not impact any existing functionality or add new functionality.

## Training
N/A

## Certification
N/A - This PR does not impact any existing functionality or add new functionality.

## Marketing
N/A
## Automation tests
 - Unit tests 
   > N/A
 - Integration tests
   > N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? N/A
 - Ran FindSecurityBugs plugin and verified report? Yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? Yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
WSO2 API Manager 2.1.0
 
## Learning
N/A

### CODE REVIEW

File: commons-beanutils/1.9.3.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~ KIND, either express or implied.  See the License for the
+ ~ specific language governing permissions and limitations
+ ~ under the License.
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>commons-beanutils.wso2</groupId>
```
Comment (thusithathilina) [inline commons-beanutils/1.9.3.wso2v1/pom.xml]: Should we change the group id to align with orbit convention?

File: geronimo-jms_1.1_spec/1.1.1.wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+    </distributionManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.geronimo.specs</groupId>
+            <artifactId>geronimo-jms_1.1_spec</artifactId>
+            <version>${org.apache.geronimo.specs.version}</version>
```
Comment (thusithathilina) [inline geronimo-jms_1.1_spec/1.1.1.wso2v1/pom.xml]: Should be optional

File: geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    </description>
+    <url>http://wso2.org</url>
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+
+        <snapshotRepository>
```
Comment (thusithathilina) [inline geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml]: remove snapshot distro

File: geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            <id>wso2.snapshots</id>
+            <name>Apache Snapshot Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/snapshots/</url>
+        </snapshotRepository>
+    </distributionManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.geronimo.specs</groupId>
+            <artifactId>geronimo-saaj_1.3_spec</artifactId>
+            <version>${geronimo.saaj.version}</version>
```
Comment (thusithathilina) [inline geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml]: should be optional

File: httpclient/4.5.3.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (thusithathilina) [inline httpclient/4.5.3.wso2v1/pom.xml]: Can we use OSGi bundle? https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient-osgi/4.5.3

File: jackson-core/2.9.1.wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,76 @@
+<!--
```
Comment (thusithathilina) [inline jackson-core/2.9.1.wso2v1/pom.xml]: jackson-core is a bundle. Can;t we directly use that?

File: jdbc-pool/9.0.0.M27.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  ~ under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.apache.tomcat.wso2</groupId>
+    <artifactId>jdbc-pool</artifactId>
+    <version>9.0.0.M27.wso2v1</version>
```
Comment (thusithathilina) [inline jdbc-pool/9.0.0.M27.wso2v1/pom.xml]: Why are we gonna depend on milestone? IMHO We should use 7.x range

File: storm-core/0.10.1.wso2v2/pom.xml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    <groupId>org.wso2.orbit.org.apache.storm</groupId>
+    <artifactId>storm-core</artifactId>
+    <version>0.10.1.wso2v2</version>
+    <packaging>bundle</packaging>
+    <name>WSO2 Carbon Orbit - Apache Storm Core</name>
+    <description>
+        This bundle will export packages from Apache Storm Core
+    </description>
+    <url>http://wso2.org</url>
+    <distributionManagement>
```
Comment (thusithathilina) [inline storm-core/0.10.1.wso2v2/pom.xml]: Add empty line

File: storm-core/0.10.1.wso2v2/pom.xml [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+    </distributionManagement>
+    <dependencies>
```
Comment (thusithathilina) [inline storm-core/0.10.1.wso2v2/pom.xml]: add empty line

File: storm-core/0.10.1.wso2v2/pom.xml [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+        <dependency>
+            <groupId>org.apache.storm</groupId>
+            <artifactId>storm-core</artifactId>
+            <version>${storm.version}</version>
+            <optional>true</optional>
+        </dependency>
+    </dependencies>
+    <build>
```
Comment (thusithathilina) [inline storm-core/0.10.1.wso2v2/pom.xml]: add empty line

File: zookeeper/3.4.9.wso2v1/pom.xml [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    <groupId>org.apache.zookeeper.wso2</groupId>
+    <artifactId>apache-zookeeper</artifactId>
+    <packaging>bundle</packaging>
+    <name>zookeeper orbit bundle</name>
+    <version>3.4.9.wso2v1</version>
+    <description>
+        This bundle will represent org.apache.zookeeper
+    </description>
+    <url>http://wso2.org</url>
+    <distributionManagement>
```
Comment (thusithathilina) [inline zookeeper/3.4.9.wso2v1/pom.xml]: Add empty lines to improve readability

File: geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml (hunk 4 above)
Comment (vidurananayakkara) [reply geronimo-saaj_1.3_spec/1.0.1.wso2v1/pom.xml]: Made `optional`

File: geronimo-jms_1.1_spec/1.1.1.wso2v1/pom.xml (hunk 2 above)
Comment (vidurananayakkara) [reply geronimo-jms_1.1_spec/1.1.1.wso2v1/pom.xml]: Made `optional`

File: jdbc-pool/9.0.1.wso2v1/pom.xml [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+    <version>9.0.1.wso2v1</version>
+    <packaging>bundle</packaging>
+    <name>tomcat jdbc-pool orbit bundle</name>
+    <description>Apache Tomcat JDBC Pool</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.tomcat</groupId>
+            <artifactId>tomcat-jdbc</artifactId>
+            <version>${tomcat.jdbc.version}</version>
```
Comment (thusithathilina) [inline jdbc-pool/9.0.1.wso2v1/pom.xml]: Should be optional

File: jdbc-pool/9.0.1.wso2v1/pom.xml [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+        </dependency>
+    </dependencies>
+
+    <repositories>
```
Comment (thusithathilina) [inline jdbc-pool/9.0.1.wso2v1/pom.xml]: Why we need wso2 repo?

File: jdbc-pool/9.0.1.wso2v1/pom.xml [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+            <url>http://maven.wso2.org/nexus/content/groups/wso2-public/</url>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>daily</updatePolicy>
+                <checksumPolicy>ignore</checksumPolicy>
+            </releases>
+        </repository>
+    </repositories>
+
+    <distributionManagement>
```
Comment (thusithathilina) [inline jdbc-pool/9.0.1.wso2v1/pom.xml]: Is this the correct distro?

File: tiles/3.0.7.wso2v1/pom.xml [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+    </distributionManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
+            <version>${javax.servlet.api.version}</version>
```
Comment (thusithathilina) [inline tiles/3.0.7.wso2v1/pom.xml]: dependencies should be optional

File: zookeeper/3.4.9.wso2v1/pom.xml [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+        <dependency>
+            <groupId>org.apache.zookeeper</groupId>
+            <artifactId>zookeeper</artifactId>
+            <version>${zookeeper.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>log4j</groupId>
+                    <artifactId>log4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (thusithathilina) [inline zookeeper/3.4.9.wso2v1/pom.xml]: should be optional

Comment (vidurananayakkara) [reply zookeeper/3.4.9.wso2v1/pom.xml]: Made optional

File: tiles/3.0.7.wso2v1/pom.xml (hunk 15 above)
Comment (vidurananayakkara) [reply tiles/3.0.7.wso2v1/pom.xml]: Made optional

File: jdbc-pool/9.0.1.wso2v1/pom.xml (hunk 12 above)
Comment (vidurananayakkara) [reply jdbc-pool/9.0.1.wso2v1/pom.xml]: Made optional

**Pull request merged**